plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -596,6 +605,15 @@ public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connect
                 }
                 // TODO (https://github.com/trinodb/trino/issues/10537) Add support for Datetime64 type
                 return Optional.of(timestampColumnMappingUsingSqlTimestampWithRounding(TIMESTAMP_MILLIS));
+
+            case Types.TIMESTAMP_WITH_TIMEZONE:
+                if (columnDataType == ClickHouseDataType.DateTime) {
+                    verify(typeHandle.getRequiredDecimalDigits() == 0, "Expected 0 as timestamp precision, but got %s", typeHandle.getRequiredDecimalDigits());
```

Comment:
nit: would probably be useful to mention that ClickHouse `DateTime` does not have sub-second precision.

(when I read this initially I thought this had something to do with pushdown etc.)

Reply:
I will add comments for this.

Reply:
> I will add comments for this.

Added.